pvlib/iotools/epw.py

```diff
@@ -226,18 +226,17 @@ def read_epw(filename, coerce_year=None):
             'Safari/537.36')})
         response = urlopen(request)
         with io.StringIO(response.read().decode(errors='ignore')) as csvdata:
-            data, meta = parse_epw(csvdata, coerce_year)
+            data, meta = _parse_epw(csvdata, coerce_year)
 
     else:
-        # Assume it's accessible via the file system
-        with open(str(filename), 'r') as csvdata:
-            data, meta = parse_epw(csvdata, coerce_year)
-
+        # Assume it's a buffer or accessible via the file system
+        with _file_context_manager(filename, 'r') as csvdata:
+            data, meta = _parse_epw(csvdata, coerce_year)
 
     return data, meta
 
 
-def parse_epw(csvdata, coerce_year=None):
+def _parse_epw(csvdata, coerce_year=None):
```

Comment:
Again it seem preferable getting rid of the parse function all together (avoids duplicating the doc string).